frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -85,8 +100,106 @@ const CatalogView = (props: CatalogViewProps) => {
 			setOpen(true);
 		}
 	}, [coursePreview, data]);
+	
+	
+	useEffect(() => {
+		if (course !== null) {	
+			if (false && legacyId !== null) {		
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+				// let temp = semester.split(' ');
+				// dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+				// axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+				// 	dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+				// 	});	
+			} else {
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+			}
+		}
+	}, [course?.courseNumber]);
+
+
+	const gradesData = useSelector((state: State) => state.grade?.gradesData ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollmentData = useSelector((state: State) => state.enrollment?.enrollmentData ?? null);
```

Comment:
Even though we may need it in the future, keeping it right now triggers rerenders when the state changes so we only add selector when we absolutely need it. Currently it is not being used.

Reply:
deleted!